Framework/Beamline/src/DetectorInfo.cpp

```diff
@@ -222,31 +186,13 @@ void DetectorInfo::setScanInterval(
     const std::pair<int64_t, int64_t> &interval) {
   checkNoTimeDependence();
   checkScanInterval(interval);
-  if (!m_scanIntervals) {
-    m_scanIntervals =
-        Kernel::make_cow<std::vector<std::pair<int64_t, int64_t>>>(
-            1, std::pair<int64_t, int64_t>{0, 1});
-  }
-  if (!m_isSyncScan) {
-    throw std::runtime_error(
-        "DetectorInfo has been initialized with a "
-        "asynchonous scan, cannot set synchronous scan interval.");
-  }
-  m_scanIntervals.access()[0] = interval;
+  m_scanIntervals[0] = interval;
```

Comment:
Related to above: Need some solution to avoid running out of sync with `ComponentInfo`.